File: leet_easy_python/leet_053/leet_053.py
class Solution(object):
    def maxSubArray(self, nums):
        prev = m = nums[0]
        sz = len(nums)
        for i in range(1,sz):
            prev = max(prev + nums[i], nums[i])
            m = max(m, prev)
        return m
        
        """
        :type nums: List[int]
        :rtype: int
        """

File: leet_easy_java/leet_053/leet_053.java
class Solution {
    public int maxSubArray(int[] nums) {
        int _max = nums[0], tmp = 0, len = nums.length;
        for(int i=0; i<len; i++){
            tmp += nums[i];
            _max = Math.max(_max, tmp);
            if(tmp<0) tmp = 0;
        }
        return _max;
    }
}